server/src/main/java/org/jetbrains/bsp/bazel/server/sync/TargetKindResolver.java

```diff
@@ -7,22 +7,28 @@
 import org.jetbrains.bsp.bazel.server.sync.model.Tag;
 
 public class TargetKindResolver {
-  private final Map<String, Tag> ruleSuffixToTargetType =
+  private static final Map<String, Set<Tag>> ruleSuffixToTargetType =
       Map.of(
-          "library", Tag.LIBRARY,
-          "binary", Tag.APPLICATION,
-          "test", Tag.TEST);
+          "library", HashSet.of(Tag.LIBRARY),
+          "binary", HashSet.of(Tag.APPLICATION),
+          "test", HashSet.of(Tag.TEST));
+
+  private static final Set<Tag> NO_IDE = HashSet.of(Tag.NO_IDE);
 
   public Set<Tag> resolveTags(TargetInfo targetInfo) {
+    if (targetInfo.getKind().equals("resources_union")) {
+      return ruleSuffixToTargetType.get("library");
+    }
+
     var tag =
         ruleSuffixToTargetType.entrySet().stream()
             .filter(entry -> targetInfo.getKind().endsWith("_" + entry.getKey()))
             .map(Map.Entry::getValue)
             .findFirst()
-            .orElse(Tag.NO_IDE);
+            .orElse(NO_IDE);
     if (targetInfo.getTagsList().contains("no-ide")) {
-      return HashSet.of(tag, Tag.NO_IDE);
+      return NO_IDE;
```

Comment:
Is it safe to drop tag? AFAIR IntelliJ is still processing such modules somehow, just not creating them in the end.

Reply:
it doesnt since we are filtering it out in `BazelProjectMapper#createModules`

Reply:
True that, but I am not sure if this is valid in terms of protocol. I just did it 'temporarly' having in mind that we might revert it if it causes any problems and solve it properly. Anyway if it happens, we can adjust this place as well.

Reply:
is this resolved? no action then?

Reply:
Maybe let's revert and have both NO_IDE and LIBRARY for example on a target. I can imagine that we will be for example implementing import depth feature and traverse the tree of deps, but only for library, binary and test targets and we could skip some transitive dep because of this.

Reply:
oh wow I didn't actually realize that I changed the logic here.  That was unintentional